planner/core/rule_join_reorder_greedy.go

```diff
@@ -110,27 +113,44 @@ func (s *joinReorderGreedySolver) constructConnectedJoinTree(tracer *joinReorder
 	return curJoinTree, nil
 }
 
-func (s *joinReorderGreedySolver) checkConnectionAndMakeJoin(leftNode, rightNode LogicalPlan) (LogicalPlan, []expression.Expression) {
+func (s *joinReorderGreedySolver) checkConnectionAndMakeJoin(leftPlan, rightPlan LogicalPlan) (LogicalPlan, []expression.Expression) {
 	var usedEdges []*expression.ScalarFunction
 	remainOtherConds := make([]expression.Expression, len(s.otherConds))
 	copy(remainOtherConds, s.otherConds)
-	for _, edge := range s.eqEdges {
+	joinType := InnerJoin
+	for idx, edge := range s.eqEdges {
 		lCol := edge.GetArgs()[0].(*expression.Column)
 		rCol := edge.GetArgs()[1].(*expression.Column)
-		if leftNode.Schema().Contains(lCol) && rightNode.Schema().Contains(rCol) {
+		if leftPlan.Schema().Contains(lCol) && rightPlan.Schema().Contains(rCol) {
+			joinType = s.joinTypes[idx]
 			usedEdges = append(usedEdges, edge)
-		} else if rightNode.Schema().Contains(lCol) && leftNode.Schema().Contains(rCol) {
-			newSf := expression.NewFunctionInternal(s.ctx, ast.EQ, edge.GetType(), rCol, lCol).(*expression.ScalarFunction)
-			usedEdges = append(usedEdges, newSf)
+		} else if rightPlan.Schema().Contains(lCol) && leftPlan.Schema().Contains(rCol) {
+			joinType = s.joinTypes[idx]
+			if joinType != InnerJoin {
+				rightPlan, leftPlan = leftPlan, rightPlan
+				usedEdges = append(usedEdges, edge)
```

Comment:
The `leftOuterJoin` and `rightOuterJoin` are same here?

Reply:
JoinNode as leftPlan or rightPlan is determined by eqConds, it can be handled uniformly